core/shared/src/main/scala/zio/Promise.scala

```diff
@@ -86,10 +86,26 @@ class Promise[E, A] private (private val state: AtomicReference[State[E, A]]) ex
   final def done(io: IO[E, A]): UIO[Boolean] = complete(io)
 
   /**
-   * Completes the promise with the specified result. If the specified promise
-   * has already been completed, the method will produce false.
+   * Completes the promise with the result of the specified effect. If the
+   * promise has already been completed, the method will produce false.
```

Comment:
Maybe add a note that `completeWith` will be much faster, so investigate using that if it makes sense?

Reply:
Done.